fastlane/lib/fastlane/actions/ensure_bundle_exec.rb

```diff
@@ -0,0 +1,61 @@
+module Fastlane
+  module Actions
+    module SharedValues
+    end
+
+    # Raises an exception and stop the lane execution if the repo is not on a specific branch
+    class EnsureBundleExecAction < Action
+      def self.run(params)
+        gemfile_path = PluginManager.new.gemfile_path
+        if gemfile_path
+          if FastlaneCore::Helper.bundler?
+            UI.success("Using bundled fastlane ✅")
+          else
+            UI.user_error!("fastlane detected a Gemfile in the current directory. however it seems like you don't use `bundle exec`. Use `bundle exec fastlane #{ARGV.join(' ')}`")
```

Comment:
How can I test this use case?

Reply:
What do you mean?

Reply:
`however` should be upper case

Reply:
I want to test scenario when I am not using `bundle exec fastlane ...`, however in my test setup I HAVE to use `bundle exec fastlane ...` to use the local fastlane repo.

Reply:
My thought was to just shell out from the tests, but `sh` only returns the test command when run during tests. Maybe with backticks?